database/database_test.go

```diff
@@ -177,6 +178,41 @@ func TestStateForDomain(t *testing.T) {
 	}
 }
 
+func TestStatesForDomains(t *testing.T) {
+	resetDB()
+
+	if err := testDB.PutState(DomainState{Name: "a.test", Status: StatusPreloaded}); err != nil {
+		t.Errorf("cannot put state %s for test StatatesForDomains", err)
+		return
+	}
+	if err := testDB.PutState(DomainState{Name: "b.test", Status: StatusPending}); err != nil {
+		t.Errorf("cannot put state %s for test StatesForDomains", err)
+		return
+	}
+
+	domainStates, err := testDB.StatesForDomains([]string{"a.test", "b.test"})
+	if err != nil {
+		t.Errorf("Cannot get states for test StatesForDomains: %s", err)
+		return
+	}
+
+	sort.Slice(domainStates, func(i, j int) bool { return domainStates[i].Name < domainStates[j].Name })
+	if len(domainStates) != 2 {
+		count := strconv.Itoa(len(domainStates))
```

Comment:
Since this is only used for the printout, you can instead do t.Errorf("Inccorect count of states for test StatesForDomains %d", len(domainStates))

Or better still, save the len(domainStates) in a "got" variable, then check against that (and the literal "2" in a "want" variable), then you can change the error message so it includes something like t.Errorf("Incorrect count of states for test StatesForDomains, got %d, wanted %d", got, want)

Reply:
Done